VHAR-6199 Tuo kohdeluettelon kohteiden kustannukset Excelistä Harjaan

jarnovayrynen opened the pull request:
-Laske Excelissä summia riveistä ja sarakkeista kaavojen avulla

jarnovayrynen committed:
VHAR-6199 Tuo päällystyskohteiden kustannukset Excelistä

jarnovayrynen committed:
Palauta ylläpitokohteet kustannusten tuonnin jälkeen

Mahdollista liitteen lataamisen yhteydessä geneerinen parametrien antaminen entisen urakka-id:n sijaan.

jarnovayrynen committed:
Päivitä ylläpitokohteet Excel-tuonnin jälkeen

jarnovayrynen committed:
Poista käyttämätön koodi paikkauskohteet:tee-excel

jarnovayrynen committed:
Käytä Excelin lataamiseen download-ikonia

jarnovayrynen committed:
Estä lukittujen kenttien muokkaaminen pääll.kust. Excelissä

jarnovayrynen committed:
Mahdollista rivin sarakkeiden summaaminen (kovakoodattu versio)

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-6199

jarnovayrynen committed:
Laske vieressäolevien Excel-solujen summat ilman INDIRECT:iä

jarnovayrynen committed:
Poista :tasaa :oikea määritykset kustannus-Excelistä

Aiheuttivat muokkaushetkeen tumman taustan.

rckopteri commented on src/clj/harja/kyselyt/yllapitokohteet.sql:
```diff
@@ -804,6 +804,23 @@ WHERE yllapitokohde = :yllapitokohde
            FROM yllapitokohde
            WHERE id = :yllapitokohde) = :urakka;
 
+-- name: tallenna-yllapitokohteen-kustannukset-yhaid!
```
Postgres sallii from lauseen käytön updatessa. Joten mä ehkä muuttaisin tämän hauin tällaiseen muotoon:
update yllapitokohteen_kustannukset 
SET kaasuindeksi = :kaasuindeksi -- ja tähän ne muutkin päivitettävät arvot
FROM yllapitokohde
WHERE
yllapitokohteen_kustannukset.yllapitokohde = yllapitokohde.id
AND yllapitokohde.urakka = :urakka-id AND yllapitokohde.yhaid = :yhaid;

jarnovayrynen replied:
Kiitos, tämäpä oli kiintoisa. Muutin noin ja lisäsin kyselylle varmuuden vuoksi testin.

jpluomajoki-solita commented on src/clj/harja/palvelin/raportointi/excel.clj:
```diff
@@ -56,14 +56,30 @@
-    (.setCellFormula cell (str "SUM(INDIRECT(ADDRESS(" (or alkurivi 1) ",COLUMN())&\":\"&ADDRESS(ROW()-1,COLUMN())))"))
-    (-> workbook
+
+(defn- evaluoi-kaava
```
Onko multimetodeihin mahdollista upottaa yhteistä toimintaa? Olisiko tämä mahdollista ajaa jokaiselle aseta-kaava! - toteutukselle ilman, että sitä tarvii laittaa sinne itse

jarnovayrynen committed:
Laske summauskaavan loppusarake vuosikohtaisesti

jarnovayrynen committed:
Mahdollista rivi-jalkeen antaminen Exceliin

Näytä määrämuutoksiin liittyvä ohjeteksti päällystysten kustannus-Excelissä rivi-jalkeen ominaisuuden avulla.

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-6199

jarnovayrynen committed:
Yhtenäistä rivi-ennen ja rivi-jalkeen luonti Excelissä

jarnovayrynen committed:
Muokkaa ylläpitokohteen tallennusta excelistä ja lisää testi

jarnovayrynen committed:
Päivitä testit viimeistä toteutusta vastaavaksi

Kaavojen syöttäminen toteutettu testin teon jälkeen

jarnovayrynen committed:
Poista turha let kohteiden summarivistä

jarnovayrynen committed:
Korjaa Excelin rivi-ennen ja rivi-jalkeen luonnit
